source/postprocess/current_surface.cc

```diff
@@ -0,0 +1,346 @@
+
+          // Remove non-unique rows from the sorted 2D vector
+          const auto last = std::unique(temp_surface.begin(), temp_surface.end(), compareRows);
+          temp_surface.erase(last, temp_surface.end());
+
+          // Resize data table.
+          TableIndices<dim-1> size_idx;
+          for (unsigned int d=0; d<dim-1; ++d)
+            size_idx[d] = temp_surface.size();
+
+          data_table.TableBase<dim-1,double>::reinit(size_idx);
+          TableIndices<dim-1> idx;
+
+          // Fill the data table with the y-values that correspond to the surface.
+          if (dim==2)
+            {
+              for (unsigned int x=0; x<(data_table.size()[0]); ++x)
+                {
+                  idx[0] = x;
+                  data_table(idx) = temp_surface[x][1];
+                }
+            }
+
+          if constexpr (dim == 2)
+            for (unsigned int x=0; x<(data_table.size()[0]); ++x)
+              data_table(x) = temp_surface[x][1];
+
+          // Fill the coordinates with the x-values used for the data table.
+          //std::array<std::vector<double>, dim-1> coordinates;
+          coordinates[0].clear();
+          for (unsigned int i=0; i<temp_surface.size(); ++i)
+            coordinates[0].push_back(temp_surface[i][0]);
+
+          // Serialize coordinates and data table in save function. In load function recreate surface function.
+          surface_function = std::make_unique<Functions::InterpolatedTensorProductGridData<dim-1>>(coordinates, data_table);
+        }
+
+      set_last_output_time (this->get_time());
+
+      // Can I override this to a void function so we don't need to print a pair?
+      const std::string filename_prefix ("update surface");
+      return std::make_pair (std::string ("Postprocess: "),
+                             filename_prefix);
+    }
+
+
+
+    template <int dim>
+    double
+    CurrentSurface<dim>::depth_including_mesh_deformation(const Point<dim> &position) const
+    {
+      AssertThrow(Plugins::plugin_type_matches<const GeometryModel::Box<dim>>(this->get_geometry_model()),
+                  ExcMessage("Depth with mesh deformation currently only works with a 2D box geometry model."));
+
+      AssertThrow(dim==2,
+                  ExcMessage("Depth with mesh deformation currently only works with a 2D geometry model."));
+
+      AssertThrow (this->get_parameters().mesh_deformation_enabled,
+                   ExcMessage("Depth with mesh deformation must be used with mesh deformation activated."));
+
+      // Convert the point to dim-1, as we aren't interested in the vertical component
+      // for the function.
+      Point<dim-1> p;
+      for (unsigned int d=0; d<dim-1; ++d)
+        p[d] = position[d];
+
+      double depth_from_surface = surface_function->value(p) - position[dim-1];
+      return std::max (depth_from_surface, 0.);
+    }
+
+
+    template <int dim>
+    void
+    CurrentSurface<dim>::declare_parameters (ParameterHandler &prm)
```

Comment:
you dont need any of the input parameters, so you can delete the `declare_parameters` and `parse_parameters` functions here and in the header file.

Reply:
Removed.